internal/registration/certs.go

```diff
@@ -0,0 +1,131 @@
+package registration
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/rand"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"encoding/pem"
+
+	"git.sr.ht/~spc/go-log"
+	"github.com/project-flotta/flotta-operator/pkg/mtls"
+)
+
+type ClientCert struct {
+	keyPath   string
+	certPath  string
+	certGroup *mtls.CertificateGroup
+}
+
+func NewClientCert(certPath string, keyPath string) *ClientCert {
+	res := &ClientCert{
+		certPath: certPath,
+		keyPath:  keyPath,
+	}
+	return res
+}
+
+func (c *ClientCert) Init() error {
+	certPem, err := os.ReadFile(c.certPath)
+	if err != nil {
+		return err
+	}
+
+	certKey, err := os.ReadFile(c.keyPath)
+	if err != nil {
+		return err
+	}
+	c.certGroup = &mtls.CertificateGroup{
+		CertPEM:    bytes.NewBuffer(certPem),
+		PrivKeyPEM: bytes.NewBuffer(certKey),
+	}
+
+	err = c.certGroup.ImportFromPem()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *ClientCert) IsRegisterCert() (bool, error) {
+	if c.certGroup == nil {
+		return false, errors.New("Certificates are not imported")
+	}
+
+	cert := c.certGroup.GetCert()
+	if cert == nil {
+		return true, errors.New("No valid cert")
+	}
+
+	return cert.Subject.CommonName == mtls.CertRegisterCN, nil
+}
+
+func (c *ClientCert) Renew(deviceID string, key crypto.PrivateKey) ([]byte, []byte, error) {
```

Comment:
It's not really clear what should be the 2 firsts return parameters - 
Maybe we can use a struct here?

Reply:
don't think about struct, named parameters for the returned value is a simpler option (which I don't think is needed here due to line 92). However, documenting the function can be helpful for that.